### PR TITLE
Fix invalid Cypher query on nested fields with  directive using  #2581

### LINKED ISSUE #2581: New @cypher argument columnName, break some query.
**Describe the bug**
New @cypher argument columnName, break some query.

**Type definitions**
```graphql
type Author {
    name: String
    mostRecentBook: Book
        @cypher(statement: "MATCH (this)-[:AUTHORED_BOOK]->(b:Book) RETURN b AS result ORDER BY b.year DESC LIMIT 1", columnName: "result")
    mostRecentBooks: [Book!]
        @cypher(statement: "MATCH (this)-[:AUTHORED_BOOK]->(b:Book) RETURN b AS result ORDER BY b.year DESC LIMIT 5", columnName: "result")
    lastPublishedYear: Int
        @cypher(statement: "MATCH (this)-[:AUTHORED_BOOK]->(b:Book) RETURN b.year AS result ORDER BY b.year DESC LIMIT 1", columnName: "result")
    books: [Book!]! @relationship(type: "AUTHORED_BOOK", direction: OUT)
}

type Book {
    name: String!
    year: Int
    refID: ID @id
    soldCopies: Int @cypher(statement: "OPTIONAL MATCH(sales:Sales) WHERE this.refID = sales.refID WITH count(sales) as result RETURN result as result", columnName: "result")
    authors: [Author!]! @relationship(type: "AUTHORED_BOOK", direction: IN)
}

type Sales {
    price: Int
    refID: ID
}
```

**To Reproduce**
```graphql
query {
  authors {
    name
    mostRecentBook {
      name
      year
      soldCopies
    }
  }
}
```

**Expected behavior**
Return the correct number of copies sold.

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Fix generated query on nested `@cypher` directives with `columnName`

## Complexity

Complexity: Low

# Issue

Closes #2581

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2587.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/tests/tck/directives/cypher.test.ts [hunk 1]
```diff
@@ -736,6 +736,7 @@ describe("Cypher directive", () => {
                     WITH this
                     CALL {
                         WITH this
+                        WITH this AS this
```
Comment (darrellwarde) [inline packages/graphql/tests/tck/directives/cypher.test.ts:739]: This needs to be aliased? Can't just be `WITH this` again?

**Review by darrellwarde — APPROVED** [review]
LGTM, just a question!

File: packages/graphql/tests/tck/directives/cypher.test.ts (hunk 1 above)
Comment (angrykoala) [reply packages/graphql/tests/tck/directives/cypher.test.ts:739]: In this particular case is not needed, but doesn't really seems to make sense to add extra logic in our codebase to treat this case any differently

**Pull request merged**